CMakeLists.txt

```diff
@@ -0,0 +1,101 @@
+cmake_minimum_required(VERSION 3.19)
+
+project(finufft VERSION 2.0.4 LANGUAGES C CXX)
+
+if(CMAKE_CXX_COMPILER_ID IN_LIST "AppleClang;Clang;GNU")
+    # Set custom compiler flags for gcc-compatible compilers
+    set(CMAKE_CXX_FLAGS_RELEASE "-O3 -DNDEBUG -funroll-loops -fcx-limited-range")
+    set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O3 -g -DNDEBUG -funroll-loops -fcx-limited-range")
+endif()
+
+include(CTest)
+
+set(FINUFFT_ARCH_FLAGS "-march=native" CACHE STRING "Compiler flags for specifying target architecture.")
+set(FINUFFT_FFTW_SUFFIX "OpenMP" CACHE STRING "Suffix for FFTW libraries (e.g. OpenMP, Threads etc.)")
+set(FINUFFT_BUILD_EXAMPLES "OFF" CACHE BOOL "Whether to build the FINUFFT examples")
+set(FINUFFT_BUILD_TESTS "OFF" CACHE BOOL "Whether to build the FINUFFT tests")
+set(FINUFFT_BUILD_FORTRAN "OFF" CACHE BOOL "Whether to build the FINUFFT Fortran examples")
+set(FINUFFT_BUILD_MATLAB "OFF" CACHE BOOL "Whether to build the FINUFFT Matlab interface")
+set(FINUFFT_ENABLE_SANITIZERS "ON" CACHE BOOL "Whether to enable sanitizers, only effective for Debug configuration.")
+
+if(FINUFFT_BUILD_MATLAB)
+    # When building for matlab, we will fetch the OpenMP library used by matlab
+    # instead of system default for compatibility.
+    find_package(Matlab REQUIRED)
+    find_library(OpenMP_iomp5_LIBRARY NAMES iomp5 HINTS ${Matlab_ROOT_DIR}/sys/os/ PATH_SUFFIXES glnxa64 maci64)
+    find_library(OpenMP_pthreads_LIBRARY NAMES pthread CMAKE_FIND_ROOT_PATH_BOTH)
+endif()
+
+# Find required packages
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+find_package(FFTW REQUIRED)
+find_package(OpenMP REQUIRED)
```

Comment:
OpenMP should be optional:
```cmake
if (ENABLE_OPENMP)
    find_package(OpenMP REQUIRED)
    list(APPEND LIBRARIES OpenMP::OpenMP_CXX)
    message(STATUS "BUILDING FINUFFT MULTITHREAD WITH OpenMP")
elseif (ENABLE_THREADS)
    find_package(Threads REQUIRED)
    list(APPEND LIBRARIES Threads::Threads)
    message(STATUS "BUILDING FINUFFT MULTITHREAD WITH Posix Threads")
else ()
    message(STATUS "BUILDING FINUFFT SINGLETHREAD")
endif ()

```

Reply:
Finufft currently has a hard dependency on OpenMP (beyond FFTW), so it is unfortunately required. We are currently also investigating pluggable threading / other threading interfaces, but it is not a big priority at the moment.

Reply:
With the makefile I can do make OMP=OFF and it generates a single threaded one though.
(a lot of warning about unknown pragmas...)
I do this in my code as I handle the threading myself

Reply:
Yes, but 
```cmake
find_package(FFTW REQUIRED)
```
does ``` FFTW_INCLUDE_DIRS ... fftw include directory paths```
So you can just:
```cmake  
target_include_directories(${target} PRIVATE "${FFTW_INCLUDE_DIRS }")
```

Reply:
Thanks for noting this! I have added a new option to build without openmp (it'll also suppress the warnings).